core-web/libs/edit-content/src/lib/fields/dot-edit-content-file-field/components/dot-form-file-editor/dot-form-file-editor.component.ts

```diff
@@ -0,0 +1,270 @@
+};
+
+@Component({
+    selector: 'dot-form-file-editor',
+    standalone: true,
+    imports: [
+        DotMessagePipe,
+        ReactiveFormsModule,
+        DotFieldValidationMessageComponent,
+        ButtonModule,
+        InputTextModule,
+        MonacoEditorModule
+    ],
+    templateUrl: './dot-form-file-editor.component.html',
+    styleUrls: ['./dot-form-file-editor.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [FormFileEditorStore]
+})
+export class DotFormFileEditorComponent implements OnInit {
+    /**
+     * Injects the FormFileEditorStore into the component.
+     *
+     * @readonly
+     * @type {FormFileEditorStore}
+     */
+    readonly store = inject(FormFileEditorStore);
+    /**
+     * A private and readonly instance of FormBuilder injected into the component.
+     * This instance is used to create and manage forms within the component.
+     */
+    readonly #formBuilder = inject(FormBuilder);
+    /**
+     * A reference to the dynamic dialog instance.
+     * This is a read-only property that is injected using the `DynamicDialogRef` token.
+     */
+    readonly #dialogRef = inject(DynamicDialogRef);
+    /**
+     * A read-only private property that holds the configuration for the dynamic dialog.
+     * This configuration is injected using the `DynamicDialogConfig` class with a generic type of `DialogProps`.
+     */
+    readonly #dialogConfig = inject(DynamicDialogConfig<DialogProps>);
+
+    /**
+     * Form group for the file editor component.
+     *
+     * This form contains the following controls:
+     * - `name`: A required string field that must match the pattern of a valid file name (e.g., "filename.extension").
+     * - `content`: An optional string field for the file content.
+     *
+     * @readonly
+     */
+    readonly form = this.#formBuilder.nonNullable.group({
+        name: ['', [Validators.required, Validators.pattern(/^[^.]+\.[^.]+$/)]],
+        content: ['']
+    });
+
+    /**
+     * Reference to the MonacoEditorComponent instance within the view.
+     * This is used to interact with the Monaco Editor component in the template.
+     *
+     * @type {MonacoEditorComponent}
+     */
+    $editorRef = viewChild.required(MonacoEditorComponent);
+
+    constructor() {
+        effect(() => {
+            const isUploading = this.store.isUploading();
+
+            if (isUploading) {
+                this.#disableEditor();
+            } else {
+                this.#enableEditor();
+            }
+        });
+
+        effect(
+            () => {
+                const isDone = this.store.isDone();
+                const uploadedFile = this.store.uploadedFile();
+
+                untracked(() => {
+                    if (isDone) {
+                        this.#dialogRef.close(uploadedFile);
+                    }
+                });
+            },
+            {
+                allowSignalWrites: true
```

Comment:
`allowSignalWrites` is not necessary, you are not modifying a signal inside the effect.

Reply:
In this case, it is necessary because when we close the dialog, the close action is read by the component that launched the dialog, and in this block, we change the state.

![Screenshot 2024-10-24 at 8 39 33 AM](https://github.com/user-attachments/assets/a5541be9-9c30-49b0-8fc6-d08d3f14daae)